### PR TITLE
Add endpoint to display controller.options

### PR DESCRIPTION
**Pull request by macartur**
Change the class controller to register all endpoints before start the APIServer.
Add the endpoint '/kytos/configuration' to render the options loaded by kytos.
Add some test for kytos/core/{controller, api_server}.py
Fix kytos/kytos-utils#86

### CODE REVIEW

File: CHANGELOG.rst
Comment (cemsbr) [inline CHANGELOG.rst]: Maybe it would be better a shorter description. Something like "- '/kytos/configuration/' endpoint to display kytos configuration.

File: kytos/core/api_server.py
Comment (cemsbr) [inline kytos/core/api_server.py]: Nice, trailing slash is the right way to do it. But where does kytos/api come from?

File: kytos/core/controller.py
Comment (cemsbr) [inline kytos/core/controller.py]: We can be less verbose and more informative with, e.g.: ``#: datetime.datetime: Time when the controller finished starting.``

File: kytos/core/controller.py
Comment (cemsbr) [inline kytos/core/controller.py]: This method has its own docstring, so you shouldn't use attribute docstring.

File: kytos/core/controller.py
Comment (cemsbr) [reply kytos/core/controller.py]: By the way, next time, try to specify the attribute type like my previous comment.

File: kytos/core/controller.py
Comment (cemsbr) [inline kytos/core/controller.py]: Not sure whether it's part of the PEP, but let's keep using no spaces between ``string`` and ``:``.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: a**n** endpoint.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: In English, adjectives have no plural form.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: return**s** all **registered endpoints.** (adjective before noun)

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: As you are not testing the endpoint results, can you simplify this test by using ``None`` as the endpoint method?

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: _working properly_ is too vague. After reading the test, I think _Server routes should include status and shutdown endpoints._ is a better description.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: Again, maybe you can just use ``None`` as the endpoint method. If not, create a new private static method instead of repeating ``def custom_endpoint...``.

File: tests/test_core/test_controller.py
Comment (cemsbr) [inline tests/test_core/test_controller.py]: ``self.assertListEqual`` gives better output on failure.

**Review by cemsbr — CHANGES_REQUESTED** [review]

File: tests/test_core/test_controller.py
Comment (diraol) [inline tests/test_core/test_controller.py]: `/`, `/static/<>` and `/index.html` are not endpoints from our REST API ....

**Review by diraol — CHANGES_REQUESTED** [review]

File: tests/test_core/test_controller.py
Comment (macartur) [reply tests/test_core/test_controller.py]: So, today these endpoints is used to serve the web-ui.

Comment (diraol) [reply tests/test_core/test_controller.py]: @macartur I know  that our web-ui is served on this URLs, but they are not REST ENDPOINTS. They are just URLs. That's what I meant. @cemsbr, what I said is reasonable?

Comment (cemsbr) [reply tests/test_core/test_controller.py]: I'm not sure whether an URL is an endpoint, but we definitely don't have only REST endpoints in this list. Using "urls" instead of "endpoints" in this method wouldn't be wrong, because, certainly, a URL can be an endpoint.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py]: My comment in the previous commit was not considered here.

File: tests/test_core/test_api_server.py
Comment (cemsbr) [inline tests/test_core/test_api_server.py:43]: My comment in the previous commit was not considered here.

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Comment by cemsbr** [comment]
@macartur, not pushing a new commit makes the review process much longer. It is OK if you want to rebase, but do it after approval.

**Comment by beraldoleal** [comment]
@macartur thanks... waiting for changes made by @diraol and @cemsbr . Also can you please rename to `/kytos/config/` ?

File: kytos/core/api_server.py
Comment (macartur) [reply kytos/core/api_server.py]: This was wrong, I removed this.

File: tests/test_core/test_api_server.py
Comment (macartur) [reply tests/test_core/test_api_server.py]: Fixed.

File: tests/test_core/test_api_server.py
Comment (macartur) [reply tests/test_core/test_api_server.py:43]: FIxed.

**Comment by macartur** [comment]
I made all changes.

**Pull request merged**
